Specification.md

```diff
@@ -0,0 +1,87 @@
+
+## Requirements
+
+### Licensing
+
+Xapps need to be licensed on an [OSI-approved license](https://opensource.org/licenses).
+
+### Operating System
+
+Xapps need to be able to run in a Linux environment. The support of other operating systems is optional.
+
+### Graphical Desktop Application
+
+This project targets graphical applications which run on desktop systems and dependencies of them. E.g. console applications, kernel-modules and system-services cannot be xapps.
+
+### Usage of standards
+
+In order to be desktop-environment agnostic, the compliance to the following [freedesktop.org](https://www.freedesktop.org) standards is mandatory to be listed as an xapp:
+
+- [Xdg Autostart](https://www.freedesktop.org/wiki/Specifications/autostart-spec/)
+- [Xdg Base Directories](https://www.freedesktop.org/wiki/Specifications/basedir-spec/)
+- [Xdg Desktop Entry](https://www.freedesktop.org/wiki/Specifications/desktop-entry-spec/)
+- [Xdg Desktop Menus](https://www.freedesktop.org/wiki/Specifications/menu-spec/)
+- [File Manager DBUS interface](https://www.freedesktop.org/wiki/Specifications/file-manager-interface/)
+- [File URI Specification](https://www.freedesktop.org/wiki/Specifications/file-uri-spec/)
+- [Free Media Player Specification](https://www.freedesktop.org/wiki/Specifications/free-media-player-specs/)
+- [Free Media Player Remote Interface Specification](https://www.freedesktop.org/wiki/Specifications/mpris-spec/)
+- [Xdg Icon Theme](https://www.freedesktop.org/wiki/Specifications/icon-theme-spec/)
+- [Xdg Icon Naming](https://www.freedesktop.org/wiki/Specifications/icon-naming-spec/)
+- [Shared mime Database](https://www.freedesktop.org/wiki/Specifications/shared-mime-info-spec/)
+- [Startup Notification](https://www.freedesktop.org/wiki/Specifications/startup-notification-spec/)
+- [Trashcan support](https://www.freedesktop.org/wiki/Specifications/trash-spec/)
+- [Bookmark Interchange Format](https://pyxml.sourceforge.net/topics/xbel/)
+
+TBD: Not sure on these:
+- [MIME Applications Associations](https://www.freedesktop.org/wiki/Specifications/mime-apps-spec/)
+- [XSettings](https://www.freedesktop.org/wiki/Specifications/xsettings-spec/)
+- [Cursor Conventions](https://www.freedesktop.org/wiki/Specifications/cursor-spec/)
+- [Sound Theme](https://www.freedesktop.org/wiki/Specifications/sound-theme-spec/)
+
+### Window decorations
+
+Applications which are part of the xapp project should support server side window decorations and use them by default. Xapps are free to optionally as well support client side decorations [^1].
+
+### Menus
+
+If menus are required for a xapps, traditional menu bars should be used by default. It is fine to optionally support e.g. hamburger menus [^2].
+
+TBD: Do we want to force a default on this?
+
+### Theming
+
+User theming has to be supported by xapps (TODO: Where is the spec for general theming?) [^3]
```

Comment:
I think this will definitely need to be fleshed out more, and maybe should be left out of the initial version of the spec.  It's unclear how an app developer would conform to this requirement.  If they're using something like GTK or Qt, then they get theming built-in.  If they aren't, then it's hard to mandate theming support when there's no general standard for it.

And even when an app is written with a toolkit that supports theming, that theming will be specific to the toolkit, not the desktop, so I'm not sure how we can really mandate anything meaningful here.

Reply:
Maybe let's put something more general in the section, to make clear that theming generally is welcome / applications should not disable theming support on purpose, if the toolkit allows so.

Suggestion: "If the used toolkit offers the possibility, xapps should be themable"

Reply:
This one seems stalled on not really having specific language to make this more than a vague suggestion. I'd suggest leaving it out for the moment so that it doesn't become a blocker. I do want this in the spec, but with wayland adoption accelerating rather quickly rather suddenly, I'd be somewhat hesitant to recommend a mechanism for applications to automatically pick up even your vague presentation preferences without creating an xkcd 927 sort of situation.

This is a great example of what I meant in my issue about it being easier to discuss things in discrete issues or PRs once we have a generally accepted spec to hang them on, even if just a draft.